### PR TITLE
Implement MS.AAD.6.1v1 check user password expiration 

### LINKED ISSUE #398: AAD Check Password Expiration
# 💡 Summary #
We currently have MS.AAD.8.1v1 ("User passwords SHALL NOT expire") marked as not implementable, but it can be checked via `Get-MgDomain.`

## Motivation and context ##
We marked it as not implementable because as far as we knew, the only way to check this programmatically was deprecated. However, yesterday someone commented on a [thread ](https://github.com/microsoftgraph/msgraph-sdk-powershell/issues/1416#issuecomment-1603501851)I had opened about a year ago with the Graph commandlet we need to do this.

## Implementation notes ##
image
image
image
image

## Acceptance criteria ##

How do we know when this work is done?

- [ ] The report output by ScubaGear correctly reflects whether or not passwords expire in the user's tenant.

### PR DESCRIPTION
**Pull request by mitchelbaker-cisa**
#  #

## 🗣 Description ##

- Import AAD settings from Get-MgBetaDomain in `ExportAADProvider.psm1`, 
- Write unit tests to indicate correct and incorrect values in `AADConfig_06_test.rego`, write functional tests in `aad.testplan.yaml`
- Implement MS.AAD.6.1v1 policy in `AADconfig.rego`.

## 💭 Motivation and context ##

 
MS.AAD.6.1v1 was marked as not implementable, but it can now be checked with `Get-MgDomain`.

This resolves #398 

## 🧪 Testing ##

From the Testing directory, run `.\RunUnitTests.ps1 -p aad -c 6`

To run functional tests, setup $TestContainers array, setup $PesterConfig to filter based on MS.AAD.6.1v1 tag, then run the following: 

`$Config = New-PesterConfiguration -Hashtable $PesterConfig`
`Invoke-Pester -Configuration $Config`

Run the tool against commercial, gcc, and gcchigh environments 

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] *All* future TODOs are captured in issues, which are referenced
      in code comments.
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

### CODE REVIEW

File: PowerShell/ScubaGear/Modules/Providers/ExportAADProvider.psm1 [hunk 1]
```diff
@@ -101,6 +101,9 @@ function Export-AADProvider {
     # Read the properties and relationships of an authentication method policy
     $AuthenticationMethodPolicy = ConvertTo-Json @($Tracker.TryCommand("Get-MgBetaPolicyAuthenticationMethodPolicy"))
 
+    # 6.1
+    $DomainSettings = ConvertTo-Json @($Tracker.TryCommand("Get-MgBetaDomain"))
```
Comment (adhilto) [inline PowerShell/ScubaGear/Modules/Providers/ExportAADProvider.psm1:105]: Remove trailing whitespace
```suggestion
    $DomainSettings = ConvertTo-Json @($Tracker.TryCommand("Get-MgBetaDomain"))
```

Comment (mitchelbaker-cisa) [reply PowerShell/ScubaGear/Modules/Providers/ExportAADProvider.psm1]: Resolved in [492c112](https://github.com/cisagov/ScubaGear/pull/708/commits/492c11257cb4c115fb615aeb1589a535838f9fa9)

**Review by adhilto — APPROVED** [review]
Looks good

File: Rego/AADConfig.rego [hunk 2]
```diff
@@ -608,17 +610,38 @@ tests contains {
 # MS.AAD.6.1v1
 #--
 
-# At this time we are unable to test for X because of Y
+# User passwords are set to not expire if they equal INT_MAX
+INT_MAX := 2147483647
```
[the remark this answers is not in the document]
Comment (mitchelbaker-cisa) [reply Rego/AADConfig.rego]: Resolved with [4b222dc](https://github.com/cisagov/ScubaGear/pull/708/commits/4b222dc8be717e9e92f4bd5af0894bf96c6f70a8)

File: Testing/Unit/Rego/AAD/AADConfig_06_test.rego [hunk 3]
```diff
@@ -1,19 +1,88 @@
...
 
 #
 # Policy MS.AAD.6.1v1
 #--
-test_NotImplemented_Correct if {
-    PolicyId := "MS.AAD.6.1v1"
 
-    Output := aad.tests with input as { }
+# User passwords are set to not expire if they equal INT_MAX
+INT_MAX := 2147483647
```
[the remark this answers is not in the document]
Comment (mitchelbaker-cisa) [reply Testing/Unit/Rego/AAD/AADConfig_06_test.rego]: Resolved [4b222dc](https://github.com/cisagov/ScubaGear/pull/708/commits/4b222dc8be717e9e92f4bd5af0894bf96c6f70a8)

File: Rego/AADConfig.rego [hunk 4]
```diff
@@ -608,17 +610,38 @@ tests contains {
...
+    "Commandlet": [ "Get-MgBetaDomain" ],
+    "ActualValue": { UserPasswordsSetToExpire, UserPasswordsSetToNotExpire },
+    "ReportDetails": DomainReportDetails(Status, UserPasswordsSetToExpire, DescriptionString),
+    "RequirementMet": Status
+} if {
+    # For the rule to pass, the user passwords for all domains shall not expire
+    # Then check if at least 1 or more domains with user passwords set to expire exist
+    DescriptionString := "domain(s) failed"
+    Conditions := [count(UserPasswordsSetToExpire) == 0, count(UserPasswordsSetToNotExpire) > 0]
+    Status := count([Condition | some Condition in Conditions; Condition == false]) == 0
```
[the remark this answers is not in the document]
Comment (mitchelbaker-cisa) [reply Rego/AADConfig.rego]: Resolved [211f70d](https://github.com/cisagov/ScubaGear/pull/708/commits/211f70da8d50bf303509024f9147778fe6f9c329)

File: Rego/Utils/AAD.rego [hunk 5]
```diff
@@ -163,4 +163,19 @@ HasAcceptableMFA(Policy) := true if {
...
+} else := false
+
+
+############################################################################
+# The report formatting functions below are for the MS.AAD.6.1v1 policy    #
+############################################################################
+
+DomainReportDetails(Status, _, _) := Description if {
+    Status == true
+    Description := "Requirement met"
```
[the remark this answers is not in the document]
Comment (mitchelbaker-cisa) [reply Rego/Utils/AAD.rego]: Resolved [211f70d](https://github.com/cisagov/ScubaGear/pull/708/commits/211f70da8d50bf303509024f9147778fe6f9c329)

**Comment by mitchelbaker-cisa** [comment]
> Updates include new rego unit tests and test run on G5 test tenant worked as expected.
> 
> Do not see any functional test to exercise this option. Is it possible to create and add one so it is included in automated functional testing as well?
> 
> See other suggested changes below.

Functional test harness updated for MS.AAD.6.1v1

File: Testing/Functional/Products/TestPlans/aad.testplan.yaml [hunk 6]
```diff
@@ -589,13 +589,59 @@ TestPlan:
         ExpectedResult: true
 
   - PolicyId: MS.AAD.6.1v1
-    TestDriver: RunScuba
+    TestDriver: RunCached
```
[the remark this answers is not in the document]
Comment (mitchelbaker-cisa) [reply Testing/Functional/Products/TestPlans/aad.testplan.yaml:592]: RunCached is used to run these tests due to potential disruptions when modifying the tenant.

**Pull request merged**